### PR TITLE
HPCC-13534 Improve support for distibuted streaming functions

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please review code, and new syntax.
@jakesmith please review changes in thor

**Comment by AttilaVamos** [comment]
Automated Smoketest Build: success
HPCC Start: OK

| setup | hthor | thor | roxie |
| ----- | ----- | ----- | ---- |
| setup | Pass | Pass | Pass |
| setup_fetch | Pass | Pass | Pass |
| setupdict | Pass | Pass | Pass |
| setupsearchindex | Pass | Pass | Pass |
| setupsq | Pass | Pass | Pass |
| setupwordindex | Pass | Pass | Pass |
| setupxml | Pass | Pass | Pass |
 

| test | hthor | thor | roxie |
| ----- | ----- | ----- | ---- |
| streamread | Pass | Pass | Pass |
 
HPCC Stop: OK

File: testing/regress/ecl/streamread.ecl [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+dsRows := doReadRows('C:\\temp\\simple');
+
+count(dsRows) = CLUSTERSIZE * 4;
+
+dataset(outRecord) doReadBlock(const varstring name) := EMBED(C++ : distributed)
+
+    static const char * rows3 = "Gavin     \x01Simon     \002Charlotte \x09TheEnd    \x00";
+
+    #body
+    //Can return constant allocations as roxie rows
```
Comment (richardkchapman) [inline testing/regress/ecl/streamread.ecl]: Is this comment really apposite here?

File: testing/regress/ecl/streamread.ecl [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        {
+        }
+
+    private:
+        Linked<IEngineRowAllocator> resultAllocator;
+        unsigned idx;
+    };
+
+    #body
+    unsigned numParts = ctx->getNodes();
```
Comment (richardkchapman) [inline testing/regress/ecl/streamread.ecl]: The distributed option on c++ implies context is passed? Or do we always pass context these days?

Comment (richardkchapman) [inline testing/regress/ecl/streamread.ecl]: How would you access these values in other embedded languages?

File: thorlcr/activities/iterate/thiterateslave.cpp [hunk 3]
```diff
@@ -418,7 +418,7 @@ class CStreamedIteratorSlaveActivity : public CSlaveActivity, public CThorDataLi
     }
     virtual void start()
     {
-        isLocal = container.queryOwnerId() && container.queryOwner().isLocalOnly();
+        bool isLocal = container.queryLocalData() || container.queryOwner().isLocalChild();
```
Comment (jakesmith) [inline thorlcr/activities/iterate/thiterateslave.cpp]: Looks sensible, but member variable 'isLocal', now no longer used and should be removed.

Comment (jakesmith) [inline thorlcr/activities/iterate/thiterateslave.cpp]: @ghalliday - no other comments.

Comment (ghalliday) [inline thorlcr/activities/iterate/thiterateslave.cpp]: That was a copy/paste error.  Should isLocal be deleted?

File: testing/regress/ecl/streamread.ecl (hunk 2 above)
Comment (ghalliday) [inline testing/regress/ecl/streamread.ecl]: In other languages you would need an ECL wrapper that passed them in as extra parameters.

File: testing/regress/ecl/streamread.ecl (hunk 1 above)
Comment (ghalliday) [inline testing/regress/ecl/streamread.ecl]: I'll clean up.

File: thorlcr/activities/iterate/thiterateslave.cpp (hunk 3 above)
Comment (jakesmith) [inline thorlcr/activities/iterate/thiterateslave.cpp]: With your change it's no longer referenced, so yes.

**Comment by ghalliday** [comment]
Closing - branch had the wrong name and got messed up.  Will reopen with fixes/

**Pull request closed without merging**